plugins/modules/gitlab_project.py

```diff
@@ -471,7 +517,13 @@ def update_project(self, project, arguments):
         for arg_key, arg_value in arguments.items():
             if arguments[arg_key] is not None:
                 if getattr(project, arg_key) != arguments[arg_key]:
-                    setattr(project, arg_key, arguments[arg_key])
+                    if arg_key == 'container_expiration_policy':
+                        old_val = getattr(project, arg_key)
+                        if all(old_val.get(key) == value for key, value in arg_value.items()):
+                            continue
+                        setattr(project, 'container_expiration_policy_attributes', arg_value)
```

Comment:
Isn't this overwriting other settings? I think you have to have a proper loop over all `arg_value.items()` and handle every single one of them, instead of using `setattr` to set everything.

Reply:
`container_expiration_policy_attributes` is a write-only attribute and as such does not have an initial value so using setattr is required. Gitlab will not change the values of subattributes that are not present in the dictionary.

unset (value=`None`) subattributes did indeed need to be removed from the final dict to prevent overriding their specific settings.